### ISSUE TITLE
Report is not being generated

### ISSUE THREAD

**Issue by adrian-arg** [extent-framework/extentreports-cucumber4-adapter#1]
Cucumber4-Adapter is not generating the desired report

I had managed to make the previous version work a few weeks ago.

All dependencies have been updated according to the information in the repository.

/* Runner Class */ 
@RunWith(Cucumber.class)
@CucumberOptions(
//		junit = "--step-notifications",
		strict = true
, features = { Environment.featuresFolder }
, glue = { Environment.stepDefinitionsFolder }
, tags = { "@Regression"}
, monochrome = true
, plugin = {"com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:",
		}
)
public class Release_Runner extends AbstractTestNGCucumberTests{
}

/* extent.properties */
extent.reporter.avent.start=true
#extent.reporter.avent.config=avent-config.xml
extent.reporter.avent.out=test-output/AventReport/

**Comment by foursyth** [comment]
Ah, I see what the issue is..  

AventReporter is not available in community version and what you see is only a place holder for users who upgrade/downgrade to/from professional edition.  Use the below properties and it will work:

```
extent.reporter.html.start=true
extent.reporter.html.config=
extent.reporter.html.out=test-output/HtmlReport/ExtentHtml.html

extent.reporter.logger.start=true
extent.reporter.logger.config=
extent.reporter.logger.out=test-output/LoggerReport/

```

I have also created small example project [here](https://github.com/foursyth/extentreports-cucumberN-example) which you can use as reference although you won't be needing it.

**Comment by adrian-arg** [comment]
> Ah, I see what the issue is..
> 
> AventReporter is not available in community version and what you see is only a place holder for users who upgrade/downgrade to/from professional edition. Use the below properties and it will work:
> 
> ```
> extent.reporter.html.start=true
> extent.reporter.html.config=
> extent.reporter.html.out=test-output/HtmlReport/ExtentHtml.html
> 
> extent.reporter.logger.start=true
> extent.reporter.logger.config=
> extent.reporter.logger.out=test-output/LoggerReport/
> ```
> I have also created small example project [here](https://github.com/foursyth/extentreports-cucumberN-example) which you can use as reference although you won't be needing it.

Thank you, foursyth.
But still no luck, the only difference now is that it creates the output folders, but the report isn't there.

extent.reporter.email.out=test-output/EmailReport/ExtentEmail.html
extent.reporter.html.out=test-output/HtmlReport/ExtentHtml.html

I double checked all dependencies and removed any unnecessary ones.

**Comment by foursyth** [comment]
Can you share the new propeties file you have? 

Having output folders is a good thing! We're almost there.

**Comment by foursyth** [comment]
One more thing. Are you running from command line or an IDE?

**Comment by adrian-arg** [comment]
I copied the one in the example project, I tried commenting the config part just in case, but same result.
I'm running in Eclipse.

extent.reporter.avent.start=false
extent.reporter.bdd.start=false
extent.reporter.cards.start=false
extent.reporter.email.start=false
extent.reporter.html.start=true
extent.reporter.klov.start=false
extent.reporter.logger.start=true
extent.reporter.tabular.start=false

extent.reporter.avent.config=
extent.reporter.bdd.config=
extent.reporter.cards.config=
extent.reporter.email.config=
extent.reporter.html.config=
extent.reporter.klov.config=
extent.reporter.logger.config=
extent.reporter.tabular.config=

extent.reporter.avent.out=test-output/AventReport/
extent.reporter.bdd.out=test-output/BddReport/
extent.reporter.cards.out=test-output/CardsReport/
extent.reporter.email.out=test-output/EmailReport/ExtentEmail.html
extent.reporter.html.out=test-output/HtmlReport/ExtentHtml.html
extent.reporter.logger.out=test-output/LoggerReport/
extent.reporter.tabular.out=test-output/TabularReport/

**Comment by foursyth** [comment]
Hmm, Eclipse works for me.

Does the example project create the reports for you?

**Comment by adrian-arg** [comment]
The example project works with Cucumber2 Adapter.

But if I change it to Cucumber4 Adapter it doesn't.

There seems to be some dependencies inconsistencies, 
If I leave the pom as:

        <dependency>
            <groupId>com.aventstack</groupId>
            <artifactId>extentreports-cucumber4-adapter</artifactId>
            <version>1.0.1</version>
        </dependency>
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-core</artifactId>
            <version>${cucumber.version}</version>
        </dependency>
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-java</artifactId>
            <version>${cucumber.version}</version>
        </dependency>
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-junit</artifactId>
            <version>${cucumber.version}</version>
            <scope>test</scope>
        </dependency>

It downloads cucumber v2.3.1 dependencies.

If I edit the POM file and bring the cucumber v4 dependencies, it doesn't work either

(Leaving aside that cucumber.api.format and others seem to be no longer available)

**Comment by foursyth** [comment]
Hmm, it should be the same. Let me try with version 4 adapter also.

**Comment by foursyth** [comment]
ello @adrian-arg 

Hope you did not have to wait for too long.  I got caught up with a few things.

Here is an example you can use:  https://github.com/foursyth/extentreports-cucumber4-example

You can see `test-output/` for the reports that got generated as per `extent.properties`.  I hope this will help you.

**Comment by adrian-arg** [comment]
Don't worry about it, I appreciate it, I'm impressed by the response time
so far.

I'm away from my computer, but at first glance I see some changes in the
runner class, I'll check it out as soon as I can. Thank you!

import org.junit.experimental.categories.Category;

@Category(Example.class)

**Comment by adrian-arg** [comment]
It seems I run into an Eclipse bug, may I ask which version are you running?

I'm on Version: Oxygen.3a Release (4.7.3a) - Build id: 20180405-1200

--
Class ParameterTypes.java >> import io.cucumber.datatable.DataTableType;

Internal compiler error: java.lang.ArrayIndexOutOfBoundsException

https://bugs.eclipse.org/bugs/show_bug.cgi?id=538192

**Comment by adrian-arg** [comment]
I tried it on another machine.

It throws a ClassNotFoundException regarding the Runner

**Comment by adrian-arg** [comment]
@foursyth 

Well, I went ahead and tried it on my project.
It seems that something is not working, I can't pin point what it is,
but depending on the feature it just plain doesn't work, it doesn't output the  stack-trace to console or generated the report.

If I run Features A and B it generates the report correctly.
If I run Feature C it doesn't generate the report.
If I run Features A, B and C it doesn't generate the report.

Could it be that Feature C has a Scenario Outline ? (even if it is not being run as it is filtered by @Tags)

The funny thing is, it does work with the old report and old version.

And to make it even more interesting, if I take out the adapter report from the runner class and use the old HTML reports, they actually work

		"pretty:Reports/TEST/StdOut/STDOUT"
		,"html:Reports/TEST/HTML"
		,"json:Reports/TESTR/JSON/TEST.json"
		,"usage:Reports/TEST/JSON/TEST-Usage.json"
		,"rerun:Reports/TEST/TEST-ReRun.txt"

**Comment by adrian-arg** [comment]
Well, I can confirm that the issue seems to be related to Features containing Scenario Outline / Examples, even If the particular Scenarios are not run.
If the Runner class references a Feature that contains Scenario Outline, then it won't generate a report.
If I take out the "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:" it generates all other HTML reports.

**Comment by foursyth** [comment]
Hello again @adrian-arg 

I do not think Scenario Outline is the culprit.  Still,  is it possible for you to share one that produces such error which we can use to test?  You do not need to provide the Java StepDefs - just the Scenario Outline.  We can create dummy step defs and test.

In `extentreports-4.0.0`, there was an issue with Asterisk (*) which was resolved in `4.0.1`.  Do any of your steps have the asterisk keyword as shown in the example below?

```
Scenario:  hello world
    Given something
    * something else
```

**Comment by foursyth** [comment]
Btw, if Asterisk was the issue, you can upgrade to version `extentreports-cucumber4-adapter:1.0.2` and it should now be resolved.  If it is something else,  it would help if you can share an example with us so we can test.

Although Scenario Outline was part of my example too, see [here](https://github.com/foursyth/extentreports-cucumber4-example/blob/master/src/test/resources/cucumber/examples/java/calculator/basic_arithmetic.feature#L17).  And it works..

**Comment by adrian-arg** [comment]
Oh, I didn't see that.
I could not run that project because of the eclipse Bug I mentioned, I'll
have to sort that out later.

I don't recall having any * in my feature files, but I do have them in my
step definitions. I will check that.

I used one of the features that worked, and added a scenario outline with a
simple:
Then display <errorMsg>
Example:
| errorMsj |
| this is an error |

Once I get home I'll look into it and let you know.

Thanks for the help!

**Comment by foursyth** [comment]
I have also noticed issues with Eclipse IDE for EE (Eclipse IDE for Java EE Developers).  The toned down IDE (Eclipse IDE for Java Developers) works fine.  I am not sure why that is so, but starting version 2 of Cucumber, my code fails to work properly in the EE version.

**Comment by deena-p** [comment]
Hi @foursyth 
I am also facing the same issue: cucumber4 adapter is not generating report (extent.reporter.html). 
I am using below versions: 
        <cucumber.type>io.cucumber</cucumber.type>
        <cucumber.version>4.0.1</cucumber.version>
        <cucumber.extent.report>1.0.2</cucumber.extent.report>

Here is my code of TestRunner:

@CucumberOptions(
        features = "src/test/java/features",
        tags = {"@AllTestScenarios"},
        plugin = {
                "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"
        }
)
public class CucumberTestRunnerWD {

    private TestNGCucumberRunner testNGCucumberRunner;

    @Before
    public void beforeSuite() {
        instantiateDriverObject();
    }

    @After
    public void afterSuite() {
        DriverFactory.closeAllDriverObjects();
    }


    @BeforeClass(alwaysRun = true)
    public void setUpClass() {
        testNGCucumberRunner = new TestNGCucumberRunner(this.getClass());
    }

    @Test(groups = "cucumber", description = "Runs Cucumber Scenarios", dataProvider = "scenarios")
    public void feature(PickleEventWrapper pickleEvent, CucumberFeatureWrapper cucumberFeature) throws Throwable {
        testNGCucumberRunner.runScenario(pickleEvent.getPickleEvent());

    }

    @DataProvider
    public Object[][] scenarios() {
        return testNGCucumberRunner.provideScenarios();
    }

    @AfterClass(alwaysRun = true)
    public void tearDownClass() {
        testNGCucumberRunner.finish();
    }
}

**Comment by foursyth** [comment]
Hello @deena-p

Would you be kind enough to send a sample with me that I can share with my team and fix asap?

**Comment by deena-p** [comment]
Hi @foursyth 
Here is the sample which I have created from my project.

https://github.com/deena-p/cucumber4-sample-deena

Thank you very much for your quick response.

**Comment by adrian-arg** [comment]
Thanks, @foursyth . I think I got it down!

Just to leave everything here in case anyone else runs into the same issue:

I'm running Eclipse IDE for Java Developers.

If I use an * in the Feature file I get a
java.lang.ClassNotFoundException:` `com.aventstack.extentreports.gherkin.model.*
The report is generated, but the particular Step is not attached.

You were right, there Scenario Outlines weren't the culprits, it just happened to be that all my failing Scenarios had a Scenario Outline.

A few hours ago I finished refactoring all of my Step Definitions so I was no longer able to reproduce the Scenario Outline issue.

So, I started messing around with the order of the Steps to force a few Exceptions and voilà!

**If the Exception is thrown in the middle of the Scenario and some Steps are Skipped, then the console doesn't show the Stack-Trace and the Report is not generated.**

It's like something is happening between the failing step and the skipped steps.

The feature file looks something like this:

  Scenario: Checkout Page
    Given an authenticated user <<< Step Fails
    When selects a product 
    Then the checkout page is displayed

  Scenario Outline: Checkout Page
    Given an authenticated user using "<e-mail>" <<< Step Fails
    When selects a product
    Then the checkout page is displayed

 Examples:
 | email                            |
 | another@user.com      |
 | someother@user.com |

__
It's very late  so I will check if the new version fixes anything tomorrow, although I see @deena-p is using the latest version.

**Comment by foursyth** [comment]
Thank you @adrian-arg and @deena-p.  I much appreciate your efforts!

**Comment by deena-p** [comment]
Hi @foursyth 
After I copied the extent.properties and .xml file to resources folder. Now it is working.
Reports are generated. Also updated the sample project.
Thank you.

**Comment by deena-p** [comment]
Hi @foursyth 
Can you please let know how to attach screenshot to the scenarios when we use the adapter?
Thank you

**Comment by foursyth** [comment]
Hello @deena-p 

Sure,  I will have a look today and let you know.

**Comment by foursyth** [comment]
Also, so its working for you now?  Wow!  

One thing I would like to point out after looking at your project.  You must use only 1 type of report from the initalization (enabled the HTML for now, as only HTML and Logger are available in community version):

```
extent.reporter.avent.start=false 
extent.reporter.bdd.start=false 
extent.reporter.cards.start=false 
extent.reporter.email.start=false 
extent.reporter.html.start=true 
extent.reporter.klov.start=false 
extent.reporter.logger.start=false 
extent.reporter.tabular.start=false 
```

**Comment by foursyth** [comment]
@deena-p Screenshots do not look so straight-forward because of the way hooks are used.  The new `EventListener` seems to be generating the report after all tests have complete, due to which, the current `ExtentTest` is not available.

**Comment by adrian-arg** [comment]
@foursyth 

I updated to the latest version and I can confirm with %99.9 certainty that:

Reports are not generated if a Scenario fails and at least one Step is Skipped.

If the Runner class runs {n} Scenarios and at least 1 fails this way, then the entire Report is not generated.

**Comment by foursyth** [comment]
In my example, I am failing a few steps and reports are getting generated.

@anshooarora

**Comment by foursyth** [comment]
@deena-p I see an embed event

https://github.com/extent-framework/extentreports-cucumber4-adapter/blob/master/src/main/java/com/aventstack/extentreports/cucumber/adapter/ExtentCucumberAdapter.java#L158

Not sure how to use this one though but looks like it gets called from the feature itself.

**Comment by foursyth** [comment]
@adrian-arg I will test again with my example with more deliberate exceptions. Hopefully that allows me to reproduce.

**Comment by adrian-arg** [comment]
> In my example, I am failing a few steps and reports are getting generated.
> 
> @anshooarora

I'm creating a new project to try and reproduce the issue, as I can't get past the Eclipse Bug regarding the DataTables import:

Errors occurred during the build.
Errors running builder 'Java Builder' on project 'extentreports-cucumber4-example-master'.
1

**Comment by foursyth** [comment]
Yeah :(

I would recommend using Intellij, that works perfectly.

**Comment by adrian-arg** [comment]
Alright, it took sometime to upload 🤣 
https://github.com/adrian-arg/extentsreport-cucumber4-example-adrian-arg

I added two runners one with:
`, tags = { "@CheckoutPage and @OK"}
`
And another with:
`, tags = { "@CheckoutPage and @BUG"}`

(Can't afford IntelligJ 💸  )

**Comment by foursyth** [comment]
LOL, me too, I use the free community version when things like the new Cucumber Ast implementation does not work.

**Comment by foursyth** [comment]
@adrian-arg 

I removed `strict = true` from CucumberOptions and I see results for both runners being created.

**Comment by adrian-arg** [comment]
@foursyth , I made a gif file

![bug](https://user-images.githubusercontent.com/32654123/48639106-59361580-e9b1-11e8-9924-4a9bae34c35f.gif)

**Comment by deena-p** [comment]
> https://github.com/extent-framework/extentreports-cucumber4-adapter/blob/master/src/main/java/com/aventstack/extentreports/cucumber/adapter/ExtentCucumberAdapter.java#L158

@foursyth @anshooarora Unless otherwise, if we don't add screenshot attachment functionality to the adapter, lot of people will not be convinced to use it. Is there any possibility to attach screenshot other than you mentioned here?

**Comment by foursyth** [comment]
@deena-p 

I wish I could answer that.  It's a much needed functionality but there is much lack of clarity around the way Cucumber 2+ listeners are created.

What I have shared is what the official cucumber formatter has, as this adapter is also a port of the HtmlFormatter directly from cucumber jvm.

We will find a way to add it and I am sure @anshooarora may get to it sooner than later, but any effort that we can put in would help get this missing piece complete.

I will create a ticket at cucumber jvm and raise this question. Please participate in the discussion there. A greater audience generally drives devs and wr may get answers soon.

**Comment by foursyth** [comment]
Here it is

https://github.com/cucumber/cucumber-jvm/issues/1502

**Comment by deena-p** [comment]
@foursyth Thank you very much for the immediate response.

**Comment by foursyth** [comment]
@anshooarora @adrian-arg @deena-p 

Please see this reply:

https://github.com/cucumber/cucumber-jvm/issues/1502#issuecomment-439597404

We need `ConcurrentEventListener`.  Ref: https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/runtime/formatter/ProgressFormatter.java#L18

**Comment by mpkorstanje** [comment]
@foursyth please don't tag me here.

**Comment by deena-p** [comment]
Hi @foursyth @anshooarora 
Are you going to provide support for this?
or As a users of your extent4 adapter, we have to take care of this by ourself. If yes, can you please help me how implement this to support screenshot attachment?

**Comment by foursyth** [comment]
@deena-p 

This adapter belongs to the community, is open-source and also driven by users like you and I.  So feel free to contribute :)   I feel we have a good starting point here and we can work together in adding the missing pieces.

**Comment by anshooarora** [comment]
Thank you for your efforts everyone.  I have a few items to work on at the moment, and hoping to start on the Cucumber adapters later this or next week.  In the meantime, please send in any pull requests - I will merge.

**Comment by anshooarora** [comment]
@adrian-arg As per @foursyth's findings, I have made several updates to the codebase.  Also completed the remaining todos (EmbedEvent).  Please test with `1.0.3` when you get a chance.

**Comment by deena-p** [comment]
Hi @anshooarora  @foursyth 
Is this support screenshot attachment functionality too?

**Comment by anshooarora** [comment]
Hello @deena-p yes, it is now supported via EmbedEvent

**Comment by anshooarora** [comment]
I am closing this. Please reopen or create a new one if issues persist.

**Comment by adrian-arg** [comment]
> @adrian-arg As per @foursyth's findings, I have made several updates to the codebase. Also completed the remaining todos (EmbedEvent). Please test with `1.0.3` when you get a chance.

Thank you, @anshooarora!
At first glance it appears to be working, I will try to do some exhaustive testing later.
Looking forward to checking out the EmbedEvent feature!
Sorry I didn't reply earlier, I am busy at work and in the middle of finals.

**Comment by deena-p** [comment]
> Hello @deena-p yes, it is now supported via EmbedEvent

@anshooarora will it support attaching screenshot to each step?

**Comment by adrian-arg** [comment]
I run a regression that I had done on Friday with no modifications other than the POM file and...
1) If a Scenario Step fails and there are still Steps to go through, the very next Step is marked as PASS, and the rest of them are not shown..
It is shown as SKIPPED in the JUNIT Step progress tab.
"cucumber.runtime.junit.SkippedThrowable: This step is skipped"
But Report shows it as PASS.
After this it doesn't go through with the code inside the @After hook (window remains open instead of performing driver.quit() )
2) The last Feature Scenario Result is PASS, but it should actually FAIL as it does when run on its own.

@anshooarora , should we re-open or create a new Issue?

**Comment by foursyth** [comment]
@adrian-arg you can reopen this, I will test a few scenarios and help in reproducing these issues.

**Comment by adrian-arg** [comment]
Thanks, @foursyth.
BTW, I edited the comment after you replied, there's a little more information on it now.

And according to Git: "You cannot re-open your own issues if a repo collaborator closed them."

**Comment by foursyth** [comment]
@anshooarora please reopen.

**Comment by foursyth** [comment]
@anshooarora 

I can confirm the 1st issue @adrian-arg shared.  However, the issue is, Cucumber is sending the events even after step has failed and it's sending passed events.  See below:

PS. Cucumber status as in caps, Extent lowercase.

```
Given today is 2011-01-20 -> PASSED, pass
When I ask if Jan 19, 2011 is in the past -> FAILED, fail  <---- failed
Then the result should be yes -> PASSED, pass  <---- this should remain as fail, however, passed
```

**Comment by foursyth** [comment]
This does not look right.  Opening a ticket with cucumber-jvm team.

**Comment by foursyth** [comment]
Reference:  https://github.com/cucumber/cucumber-jvm/issues/1505

**Comment by foursyth** [comment]
@anshooarora I have figured out the issue.  The issue is that in such a case a Throwable is not created and thus, the method `skip(throwable)` receive a NULL, which causes a NullPointer.  I suggest changing this:

```
stepTestThreadLocal.get().skip(result.getError());
```

**Comment by anshooarora** [comment]
Thank you.  Its a little difficult for me to do all this without your help.

Can you check with `1.0.4-SNAPSHOT` and let me know if all works well now? [link](https://oss.sonatype.org/content/repositories/snapshots/com/aventstack/extentreports-cucumber4-adapter/1.0.4-SNAPSHOT/)

```
<dependency>
  <groupId>com.aventstack</groupId>
  <artifactId>extentreports-cucumber4-adapter</artifactId>
  <version>1.0.4-SNAPSHOT</version>
</dependency>
```

**Comment by foursyth** [comment]
Ok, with this snapshot, all my tests which were giving invalid results have magically fixed themselves LOL.  But I am EXHAUSTED now and will do thorough tests tomorrow.

@adrian-arg @deena-p If you don't mind giving a helping hand, could you also check with the snapshot @anshooarora has shared just now?

**Comment by foursyth** [comment]
@anshooarora and I ran several tests and all status coming in the correct format. 

I have also tested EmbedEvent, works beautifully!  Here is how you use it:

1. Any image you capture, convert into byte array:
```
public byte[] extractBytes (String imageName) throws IOException {
    return Files.readAllBytes(new File(imageName).toPath());
}
```

2. Use `@After` or `@AfterStep` to embed:
```
@After
public void after(Scenario scenario) {
    scenario.embed(extractBytes("img.png"), "image/png");
}
```
```
@AfterStep
public void afterStep(Scenario scenario) {
    scenario.embed(extractBytes("img.png"), "image/png");
}
```

@anshooarora just committed c1aa9f568b797f02930a657dd21c385a00fb5c69.  There are also a few very nice updates to the look and feel.  We will now release extentreports with the updates, then this adapter.  Hopefully it works for you!  Cheers and Happy Thanksgiving.

**Comment by anshooarora** [comment]
`extentreports:4.0.5` deployed to central
`extentreports-cucumber4-adapter:1.0.4` deployed to central

**Comment by anshooarora** [comment]
I am closing this.  Please open new individual tickets according to the issues or questions that now arise.

**Comment by adrian-arg** [comment]
Sorry I'm late to the party! I'll be mostly away from my computer for a few days.

I run the same tests as last time, unfortunately I didn't get any Skipped Steps, I will test that at a later time.

Regarding the other issue I found:

2) The last Feature Scenario Result is PASS, but it should actually FAIL as it does when run on its own.

I'm seeing that the report for the Failure of last Feature tested is being added to the second to last Feature.

The last Feature is reported as PASS but empty

1st Image
![image](https://user-images.githubusercontent.com/32654123/48920705-2ee4cc00-ee79-11e8-897a-3d62f05e7d25.png)

2nd Image
![last feature](https://user-images.githubusercontent.com/32654123/48920719-3e641500-ee79-11e8-9b4b-9bd98adb57b5.png)

**Comment by foursyth** [comment]
Hello @adrian-arg

Understood, thanks for helping out!  I guess I need to run more tests but the ones I have run today work flawlessly.  There are several changes to the way tests now run with cucumber-4,  I would need to see how you have implemented those.

Whenever you get a chance, no rush.  Hopefully others using this adapter may also report similar issues.  

We don't really use Cucumber and only creating this adapter for the community. So it's been a lot of learning, especially the intricacies of the new event driven system.

From this point on, it would help to have real examples to analyze and debug issues as the scenarios we created and ported over from the official cucumber repo all work without issues.

**Comment by adrian-arg** [comment]
Hello, @foursyth , your work is much appreciated!

I run into the old post claiming Cucumber v2 didn't work anymore just recently, I actually was using Cucumber 3, and didn't use Extent report before.
But the project demanded a more readable report, and after a looooong struggle trying to make it work I downgraded everything to Cucumber 1.2.5 .
The report is working but no DataTable types and the Screenshots couldn't not be exported so I was glad to see you guys were working on an adapter so soon (at least for me :D )

I updated the example project:

https://github.com/adrian-arg/extentsreport-cucumber4-example-adrian-arg

The { junit = "--step-notifications" } in @CucumberOptions will help you see the JUnit Steps details that stopped working with the more recent versions of Cucumber.

As you will see, the last Feature run (0003_CheckoutPage) Fails, but in the report it appears Blank and PASS.
If you see the end of the previous Feature (0002_CheckoutPage) in the Report, you will see that it was attached there.

![image](https://user-images.githubusercontent.com/32654123/48924702-d83bba00-ee99-11e8-9fb6-c26fd28c061c.png)

---------------------

I'm also seeing that in the first Feature (0001_CheckoutPage)
The similarity in the Scenario Outline names (they match a prefix until a dash).

  Scenario Outline: Checkout Page for Other Users - Bugged Scenario
  Scenario Outline: Checkout Page for Other Users - OK

And that causes the report to output a Single Scenario Outline under the name of the first:

ScenarioOutline: Checkout Page for Other Users - Bugged Scenario

__________________

I will try to help more as soon as I'm done with my courses.

Sorry for the naming, I will try to give all the features and Scenarios proper names so that they can be more easily identifiable when I have more time on my hands.

Although, doing silly rushed naming like that caused me to find the last thing I described here.

_________________
PS: I won't say I'm an expert on Cucumber but I got quite familiar with the tool, if there´s anything I can help with, please let me know.

**Comment by deena-p** [comment]
Sorry for the late response. @foursyth @anshooarora 
As I mostly concentrate on screenshot attachment functionality, here are my first observations.
1. This extent-report is not portable with screenshot attached. Is that possible to our provide own server location to store screenshots?
   Currently it is hard coded.
![image](https://user-images.githubusercontent.com/12594662/48930673-22fc1700-ef18-11e8-99c5-f22708a24a96.png)
  Can we change it to something like this.
![image](https://user-images.githubusercontent.com/12594662/48933951-226b7c80-ef28-11e8-8505-600d1c7affe4.png)

2. Instead of showing whole screeshot picture completely in the report, is that possible to provide only link of the sreenshot ( like a icon) in the report?

**Comment by deena-p** [comment]
My third concern, @foursyth @anshooarora 
In case if I want to display all the scenarios on the left hand side of the dashboard page where currently feature file name are displaying? is that possible?
![image](https://user-images.githubusercontent.com/12594662/48927867-be34c280-eeff-11e8-992e-37baadfa4224.png)

See below the report when we run them in parallel mode, you can see same feature name is displayed again and again. Instead if we allow displaying of scenario names there, the report will look more meaningful.
![image](https://user-images.githubusercontent.com/12594662/48940606-50f45200-ef3e-11e8-8726-bd95900a8364.png)

**Comment by foursyth** [comment]
@adrian-arg thank you,  I have forked your project just now and will test.

@deena-p 

 * can be easily done with `extent.properties` or code.  Can you send a pull request for @anshooarora to review?
 *  sure, do this via a mix of JS and CSS from your html-config.xml,  or if you only want to reduce the style use the CSS config
 * No, this is not possible, and not recommended either..  where will your features be displayed?

**Comment by foursyth** [comment]
@adrian-arg I am able to reproduce..  my assumption is that its because the name of the scenario is the same between the 2 feature files.  Maybe checking for feature name as well would alleviate this issue.

@anshooarora I am rereferring to this https://github.com/extent-framework/extentreports-cucumber4-adapter/blob/master/src/main/java/com/aventstack/extentreports/cucumber/adapter/ExtentCucumberAdapter.java#L250

**Comment by foursyth** [comment]
@anshooarora please reopen this again :)

**Comment by anshooarora** [comment]
@adrian-arg I just worked on your example with foursyth - this should be fixed.  Please try with:

```
<version>1.0.5-SNAPSHOT</version>
```

[link](https://oss.sonatype.org/content/repositories/snapshots/com/aventstack/extentreports-cucumber4-adapter/1.0.5-SNAPSHOT/)

The issue was that the `scenarioOutline` ThreadLocal wasn't being reset after completion.

**Comment by anshooarora** [comment]
@deena-p Do you mind opening a new ticket for the 1st item?  As foursyth said, this can be added via config (properties or Java, although I prefer properties).  Let me know your thoughts as well.

**Comment by deena-p** [comment]
@anshooarora @foursyth 
Hi I have created a new ticket for the first item. #2

**Comment by reach2jeyan** [comment]
@foursyth 
Can you explain why do I need to use `extractBytes` when I am able to have the screenshots attached to the extent reports with 

```
TakesScreenshot ts = (TakesScreenshot) Main.driver;
            byte[] screenshot = ts.getScreenshotAs(OutputType.BYTES);
            scenario.embed(screenshot, "image/png");
```

I never needed to use `scenario.embed(extractBytes(name));`
![screnshotextent](https://user-images.githubusercontent.com/20741047/49575825-aa5e7700-f969-11e8-9cd6-f8586f1f2790.PNG)

**Comment by anshooarora** [comment]
@reach2jeyan see [here](https://github.com/cucumber/cucumber-jvm/blob/master/core/src/main/java/cucumber/api/Scenario.java#L41) for reference, as the only possible way to do this is via a byte array.

```
void embed(byte[] data, String mimeType);
```

**Comment by anshooarora** [comment]
Hello @reach2jeyan if your existing method returns a byte array, then you do not need to use the method recommended by @foursyth.

**Comment by testdgvn** [comment]
Hi @reach2jeyan can you please show me in detail your embed method? I've tried yours but got no luck seeing the screenshots attached to report (screenshots is just generated to proper folder instead). Where did you implement the method? 

I'm using extentreports-cucumber4-adapter: 1.0.7 and extentreports: 4.0.9

Thanks in advance!
cc @anshooarora

**Comment by foursyth** [comment]
As shared earlier:

```
@After
public void after(Scenario scenario) {
    scenario.embed(extractBytes("img.png"), "image/png");
}

@AfterStep
public void afterStep(Scenario scenario) {
    scenario.embed(extractBytes("img.png"), "image/png");
}
```

**Comment by subha077** [comment]
Hi, @foursyth , @anshooarora ..
Great work on the adapter 4. I am able to generate detailed report now. But Now I am facing two issues.

(1) Can not add system info to the report - please suggest.
(2) The error screenshots are being generated but not getting attached to the extent report. I tried to embed them using @After method.

Here is the link for the sample project. can you please suggest and advice as soon as possible.

Thanks in advance..
Here is the link for the [project code.](https://github.com/subha077/AutomationSubhankarRepo/tree/master/CucumberWithPOM)

**Comment by Andyclark97** [comment]
@foursyth @anshooarora I am having same problem as @subha077 above. Screenshots are being generated by not added to the report.

**Comment by igorakintev** [comment]
I have the same bug as @subha077 and @Andyclark97

**Comment by MansiVyas4** [comment]
Hello, I am using io.cucumber version 4.3.1 and need to add extent report to my project.
So, I have adding cucumber adapter version 1.0.7  for extent reports. But, report is not getting generated.
I have added cucumber and its depenedcy with extent cucumber adapter. I am running my project through cmd adding plugin as -p com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter: 
Also, placed my extent-config in src/test/resource. 
Only this thing needs to be done. Right???? Or any other things to be configured or code needs to be written?????

**Comment by subha077** [comment]
@Mansi- I believe you are talking about cucumber core v4.3.1.
If you are using a maven project, just add the below dependency. It will populate all other requirements regarding cucumber jars as maven dependency.

<dependency>
    <groupId>com.aventstack</groupId>
    <artifactId>extentreports-cucumber4-adapter</artifactId>
    <version>1.0.7</version>
</dependency>


Try and let us know. I could generate the detailed report but having some issues as I mentioned above.

**Comment by SamiraKhodami** [comment]
@foursyth @anshooarora I am having same problem as @subha077 above. Screenshots are being generated but not added to the report.
I am using extentreports-cucumber4-adapter , version 1.0.7 with cucumber 4.2.0

**Comment by Vinoth0194** [comment]
Hi, Can some One help me out to fix this issue?

io.cucumber.core.exception.CucumberException: Couldn't load plugin class: com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter

I am getting this error
This is my runner class.
<img width="1440" alt="Screenshot 2020-06-24 at 12 28 34 PM" src="https://user-images.githubusercontent.com/58105225/85511433-3b617580-b616-11ea-8ea6-27dd880dc59f.png">

and this is my pom.XML
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>org.yoco.example</groupId>
    <artifactId>YoCoS</artifactId>
    <version>1.0-SNAPSHOT</version>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                </configuration>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.0.0-M5</version>
                <configuration>



                </configuration>
            </plugin>
        </plugins>
    </build>

    <dependencies>

        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>3.141.59</version>
        </dependency>


        
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.13</version>
            <scope>test</scope>
        </dependency>

        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-java</artifactId>
            <version>6.1.1</version>
        </dependency>



        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-junit</artifactId>
            <version>6.1.1</version>
            <scope>test</scope>
        </dependency>

        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-core</artifactId>
            <version>6.1.1</version>
        </dependency>


        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-jvm-deps</artifactId>
            <version>1.0.6</version>
            
        </dependency>

        <dependency>
            <groupId>net.masterthought</groupId>
            <artifactId>cucumber-reporting</artifactId>
            <version>5.3.0</version>
        </dependency>

        <dependency>
            <groupId>com.github.javafaker</groupId>
            <artifactId>javafaker</artifactId>
            <version>1.0.2</version>
        </dependency>

        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-picocontainer</artifactId>
            <version>5.7.0</version>
            <scope>test</scope>
        </dependency>







        <dependency>
            <groupId>com.aventstack</groupId>
            <artifactId>extentreports-cucumber4-adapter</artifactId>
            <version>1.0.12</version>
            <exclusions>
                <exclusion>
                    <groupId>io.cucumber</groupId>
                    <artifactId>cucumber-java</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>io.cucumber</groupId>
                    <artifactId>cucumber-core</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

    </dependencies>


</project>

**Comment by anshooarora** [comment]
@Vinoth0194 This is the version4 adapter, not tested with versions 5+.

**Comment by anshooarora** [comment]
@subha077 @Andyclark97 @igorakintev @SamiraKhodami 

Please let me know if you are still having this issue.

**Comment by Vinoth0194** [comment]
@anshooarora I am not getting it.

**Comment by anshooarora** [comment]
You have:

```
    <dependency>
        <groupId>io.cucumber</groupId>
        <artifactId>cucumber-java</artifactId>
        <version>6.1.1</version>
    </dependency>
```

with:

```
    <dependency>
        <groupId>com.aventstack</groupId>
        <artifactId>extentreports-cucumber4-adapter</artifactId>
        <version>1.0.12</version>
        <exclusions>
            <exclusion>
                <groupId>io.cucumber</groupId>
                <artifactId>cucumber-java</artifactId>
            </exclusion>
            <exclusion>
                <groupId>io.cucumber</groupId>
                <artifactId>cucumber-core</artifactId>
            </exclusion>
        </exclusions>
    </dependency>
```

Version 4 adapter with Cucumber 6. Not supported.

**Comment by Vinoth0194** [comment]
okay. I will change the version and check the reports.

**Comment by Vinoth0194** [comment]
I have changed the version: but now i am getting this error

java.lang.NoClassDefFoundError: io/cucumber/core/eventbus/EventBus

	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getConstructor0(Class.java:3075)
	at java.lang.Class.getConstructor(Class.java:1825)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.internal.requests.ClassRequest.createRunner(ClassRequest.java:28)
	at org.junit.internal.requests.MemoizingRequest.getRunner(MemoizingRequest.java:19)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:49)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: java.lang.ClassNotFoundException: io.cucumber.core.eventbus.EventBus
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 15 more

Updated POM.xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>org.yoco.example</groupId>
    <artifactId>YoCoS</artifactId>
    <version>1.0-SNAPSHOT</version>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                </configuration>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.0.0-M5</version>
                <configuration>



                </configuration>
            </plugin>
        </plugins>
    </build>

    <dependencies>

        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>3.141.59</version>
        </dependency>


        
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.13</version>
            <scope>test</scope>
        </dependency>

        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-java</artifactId>
            <version>4.8.1</version>
        </dependency>



        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-junit</artifactId>
            <version>6.1.1</version>
            <scope>test</scope>
        </dependency>

        
        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-core</artifactId>
            <version>4.8.1</version>
        </dependency>


        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-jvm-deps</artifactId>
            <version>1.0.6</version>
            
        </dependency>







        <dependency>
            <groupId>com.github.javafaker</groupId>
            <artifactId>javafaker</artifactId>
            <version>1.0.2</version>
        </dependency>

        <dependency>
            <groupId>io.cucumber</groupId>
            <artifactId>cucumber-picocontainer</artifactId>
            <version>4.8.1</version>
            <scope>test</scope>
        </dependency>







        <dependency>
            <groupId>com.aventstack</groupId>
            <artifactId>extentreports-cucumber4-adapter</artifactId>
            <version>1.0.12</version>
            <exclusions>
                <exclusion>
                    <groupId>io.cucumber</groupId>
                    <artifactId>cucumber-java</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>io.cucumber</groupId>
                    <artifactId>cucumber-core</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

    </dependencies>


</project>

**Comment by Vinoth0194** [comment]
thanks. Its working fine.

**Comment by SonaliSihra** [comment]
@anshooarora  Hey, with all the discussion above it is getting confusing. I'm facing the issue where the reports are getting generated but the images are not found in the report... I figured out that the path is set wrong in the report.html file and it is searching for the reports at http://localhost:63342/home/User/locustapitest/target/cucumber-reports/locustcharts/performanceChart1594459662333.png
whereas the image is available on http://localhost:63342/locustapitest/target/cucumber-reports/locustcharts/performanceChart1594459662333.png. 
I think I have a problem with cucumber and com.aventstack version in my pom.xml Can you please confirm once which com.aventstack version is suitable for 201.7846.6 (plugin in intellij) , Currently I'm using 		
                <dependency>
    		<groupId>com.vimalselvam</groupId>
    		<artifactId>cucumber-extentsreport</artifactId>
    		<version>3.1.1</version>
		</dependency>
		
		<dependency>
    		<groupId>com.aventstack</groupId>
    		<artifactId>extentreports</artifactId>
    		<version>3.1.5</version>
		</dependency>

**Comment by anshooarora** [comment]
@SonaliSihra  the dependencies you are using seem correct, but I believe these are for version 1 of Cucumber.  Is this the version you are using?  You can find the correct adapters from this page:  https://github.com/extent-framework/extentreports-java/wiki/Plugins

**Comment by praveen1626** [comment]
> Hi, Can some One help me out to fix this issue?
> 
> io.cucumber.core.exception.CucumberException: Couldn't load plugin class: com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter
> 
> I am getting this error
> This is my runner class.
> <img alt="Screenshot 2020-06-24 at 12 28 34 PM" width="1440" src="https://user-images.githubusercontent.com/58105225/85511433-3b617580-b616-11ea-8ea6-27dd880dc59f.png">
> 
> and this is my pom.XML
> 
> 
> 4.0.0
> org.yoco.example
> YoCoS
> 1.0-SNAPSHOT
> 
> ```
> <build>
>     <plugins>
>         <plugin>
>             <groupId>org.apache.maven.plugins</groupId>
>             <artifactId>maven-compiler-plugin</artifactId>
>             <version>3.8.1</version>
>             <configuration>
>                 <source>1.8</source>
>                 <target>1.8</target>
>             </configuration>
>         </plugin>
> 
>         <plugin>
>             <groupId>org.apache.maven.plugins</groupId>
>             <artifactId>maven-surefire-plugin</artifactId>
>             <version>3.0.0-M5</version>
>             <configuration>
> ```
> 
> ```
>             </configuration>
>         </plugin>
>     </plugins>
> </build>
> 
> <dependencies>
> 
>     <dependency>
>         <groupId>org.seleniumhq.selenium</groupId>
>         <artifactId>selenium-java</artifactId>
>         <version>3.141.59</version>
>     </dependency>
> 
> 
>     
>     <dependency>
>         <groupId>junit</groupId>
>         <artifactId>junit</artifactId>
>         <version>4.13</version>
>         <scope>test</scope>
>     </dependency>
> 
>     
>     <dependency>
>         <groupId>io.cucumber</groupId>
>         <artifactId>cucumber-java</artifactId>
>         <version>6.1.1</version>
>     </dependency>
> 
> 
> 
>     
>     <dependency>
>         <groupId>io.cucumber</groupId>
>         <artifactId>cucumber-junit</artifactId>
>         <version>6.1.1</version>
>         <scope>test</scope>
>     </dependency>
> 
>     
>     <dependency>
>         <groupId>io.cucumber</groupId>
>         <artifactId>cucumber-core</artifactId>
>         <version>6.1.1</version>
>     </dependency>
> 
> 
>     <dependency>
>         <groupId>io.cucumber</groupId>
>         <artifactId>cucumber-jvm-deps</artifactId>
>         <version>1.0.6</version>
>         
>     </dependency>
> 
>     <dependency>
>         <groupId>net.masterthought</groupId>
>         <artifactId>cucumber-reporting</artifactId>
>         <version>5.3.0</version>
>     </dependency>
> 
>     <dependency>
>         <groupId>com.github.javafaker</groupId>
>         <artifactId>javafaker</artifactId>
>         <version>1.0.2</version>
>     </dependency>
> 
>     <dependency>
>         <groupId>io.cucumber</groupId>
>         <artifactId>cucumber-picocontainer</artifactId>
>         <version>5.7.0</version>
>         <scope>test</scope>
>     </dependency>
> ```
> 
> ```
>     <dependency>
>         <groupId>com.aventstack</groupId>
>         <artifactId>extentreports-cucumber4-adapter</artifactId>
>         <version>1.0.12</version>
>         <exclusions>
>             <exclusion>
>                 <groupId>io.cucumber</groupId>
>                 <artifactId>cucumber-java</artifactId>
>             </exclusion>
>             <exclusion>
>                 <groupId>io.cucumber</groupId>
>                 <artifactId>cucumber-core</artifactId>
>             </exclusion>
>         </exclusions>
>     </dependency>
> 
> </dependencies>
> ```

Hi Vinoth,

Even i am facing the exact issue here.. I see you have fixed the issue, can you please help me here to update my POM xml..

Please find the POM.xml of mine

<dependencies>
    
    <dependency>
      <groupId>io.cucumber</groupId>
      <artifactId>cucumber-java</artifactId>
      <version>5.7.0</version>
    </dependency>
    <dependency>
      <groupId>io.cucumber</groupId>
      <artifactId>cucumber-core</artifactId>
      <version>5.7.0</version>
    </dependency>
    
    <dependency>
      <groupId>io.cucumber</groupId>
      <artifactId>cucumber-junit</artifactId>
      <version>5.7.0</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.cucumber</groupId>
      <artifactId>cucumber-testng</artifactId>
      <version>5.7.0</version>
    </dependency>

    
    <dependency>
      <groupId>org.seleniumhq.selenium</groupId>
      <artifactId>selenium-java</artifactId>
      <version>3.141.59</version>
    </dependency>
    <dependency>
      <groupId>org.testng</groupId>
      <artifactId>testng</artifactId>
      <version>6.14.3</version>
    </dependency>
<dependency>
      <groupId>com.aventstack</groupId>
      <artifactId>extentreports-cucumber4-adapter</artifactId>
      <version>1.0.12</version>
      <exclusions>
        <exclusion>
          <groupId>io.cucumber</groupId>
          <artifactId>cucumber-java</artifactId>
        </exclusion>
        <exclusion>
          <groupId>io.cucumber</groupId>
          <artifactId>cucumber-core</artifactId>
        </exclusion>
      </exclusions>
    </dependency>



**_### Error Message:_**

io.cucumber.core.exception.CucumberException: Couldn't load plugin class: com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter

	at io.cucumber.core.options.PluginOption.loadClass(PluginOption.java:117)
	at io.cucumber.core.options.PluginOption.parsePluginName(PluginOption.java:102)
	at io.cucumber.core.options.PluginOption.parse(PluginOption.java:86)
	at io.cucumber.core.options.RuntimeOptionsBuilder$ParsedPluginData.addPluginName(RuntimeOptionsBuilder.java:238)
	at io.cucumber.core.options.RuntimeOptionsBuilder.addPluginName(RuntimeOptionsBuilder.java:62)
	at io.cucumber.core.options.CucumberOptionsAnnotationParser.addPlugins(CucumberOptionsAnnotationParser.java:109)
	at io.cucumber.core.options.CucumberOptionsAnnotationParser.parse(CucumberOptionsAnnotationParser.java:57)
	at io.cucumber.testng.TestNGCucumberRunner.<init>(TestNGCucumberRunner.java:79)
	at io.cucumber.testng.AbstractTestNGCucumberTests.setUpClass(AbstractTestNGCucumberTests.java:22)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:59)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:458)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:222)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:142)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:168)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:105)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: cucumber/api/event/ConcurrentEventListener
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at io.cucumber.core.options.PluginOption.loadClass(PluginOption.java:110)
	... 22 more
Caused by: java.lang.ClassNotFoundException: cucumber.api.event.ConcurrentEventListener
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)

**Comment by reshmi537** [comment]
Report is not generating.But no error.Could u help me?

**Comment by saravanakumarm1986** [comment]
Above issue, i am also facing. Please give me a possible solution to resolve this
